azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub.py

```diff
@@ -35,7 +35,7 @@ def _run_op(self, op):
             self.run_ops_serial(
                 pipeline_ops_iothub.SetAuthProviderArgs(
                     device_id=auth_provider.device_id,
-                    module_id=auth_provider.module_id,
+                    module_id=getattr(auth_provider, "module_id", None),
```

Comment:
Making a new comment since the old one got squashed - I understand defaulting to None for defensive purposes, but this is only important if the auth_provider might not have a module_id attribute, which it always does. The "optional" part of the module_id attribute just means it may or may not be set to None. But the point is, the attribute is always there.

Using getattr is only needed if there's some kind of weird class hierarchy where some auth_providers may or may not have the attribute (but here, all auth providers have the attribute because it comes from the Abstract Class that serves as the super to all auth providers)